asllib/AST.mli

```diff
@@ -237,7 +237,8 @@ type lexpr_desc =
   | LE_Slice of lexpr * slice list
   | LE_SetArray of lexpr * expr
   | LE_SetField of lexpr * identifier
-  | LE_SetFields of lexpr * identifier list
+  | LE_SetFields of lexpr * identifier list * (int * int) list
+      (** LE_SetFields (le, fields, _) unpacks the various fields. Second argument is a type annotation. *)
```

Comment:
Argument of type `identifier list`  being a type annotation does not look immediate to me.